_data-prepper/pipelines/configuration/sources/documentdb.md

```diff
@@ -0,0 +1,98 @@
+`direct_connection` | No | Boolean  | When `true`, the MongoDB driver connects directly to the specified DocumentDB server(s) without discovering and connecting to the entire replica set. Defaults to `true`.
+`read_preference` | No | String  | Determines how to read from DocumentDB. See [Read Preference Modes](https://www.mongodb.com/docs/v3.6/reference/read-preference/#read-preference-modes) for details. Defaults to `primaryPreferred`.
+`disable_s3_read_for_leader` | No | Boolean  | When `true`, the current leader node does not read from Amazon S3. It only reads the stream. Defaults to `false`.
+`partition_acknowledgment_timeout` | No | Duration  | Configures the duration for which the node holds a partition. Defaults to `2h`.
+`acknowledgments` | No | Boolean  | When set to `true`, enables [end-to-end acknowledgments]({{site.url}}{{site.baseurl}}/data-prepper/pipelines/pipelines#end-to-end-acknowledgments) on this source after events are sent to the sinks.
+`insecure` | No | Boolean | Disables TLS. Defaults to `false`. Do not use this value in production.
+`ssl_insecure_disable_verification` | No | Boolean | Disables TLS hostname verification. Defaults to `false`. Do not enable this flag in production. Instead, use the `trust_store_file_path` to verify the hostname.
+
+### authentication
+
+The following parameters enable you to configure authentication for the DocumentDB cluster.
+
+Option | Required | Type | Description
+:--- | :--- | :--- | :---
+`username` | Yes | String | The username to use when authenticating with the DocumentDB cluster. Supports automatic refresh.
+`password` | Yes | String | The password to use when authenticating with the DocumentDB cluster. Supports automatic refresh.
+
+
+### collection
+
+The following parameters enable you to configure the collection to read from the DocumentDB cluster.
+
+Option | Required | Type | Description
+:--- | :--- | :--- | :---
+`collection` | Yes | String | The name of the collection
+`export` | No | Boolean | Whether to include an export or full load. Defaults to `true`.
+`stream` | No | Boolean | Whether to enable a stream. Defaults to `true`.
+`partition_count` | No | Integer | Defines the number of partitions to create in Amazon S3. Defaults to `100`.
+`export_batch_size` | No | Integer | Defaults to `10,000`.
+`stream_batch_size` | No | Integer | Defaults to `1,000`.
+
+## aws
+
+The following parameters enable you to configure your access to Amazon DocumentDB.
+
+Option | Required | Type | Description
+:--- | :--- | :--- | :---
+`sts_role_arn` | No | String | The AWS Security Token Service (AWS STS) role to assume for requests to Amazon SQS and Amazon S3. Defaults to `null`, which uses the [standard SDK behavior for credentials](https://docs.aws.amazon.com/sdk-for-java/latest/developer-guide/credentials.html).
```

Comment:
Should we link to the AWS service pages?

Reply:
@dlvenable Please advise.